oauth-client-util/src/main/java/org/radarcns/oauth/OAuth2AccessToken.java

```diff
@@ -0,0 +1,191 @@
+package org.radarcns.oauth;
+
+/*
+ * Copyright 2017 King's College London
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import okhttp3.Response;
+import org.radarcns.exception.TokenException;
+
+import java.io.IOException;
+import java.time.Instant;
+
+public class OAuth2AccessToken {
+
+    @JsonProperty("access_token")
+    private final String accessToken;
+
+    @JsonProperty("token_type")
+    private final String tokenType;
+
+    @JsonProperty("expires_in")
+    private final long expiresIn;
+
+    @JsonProperty("scope")
+    private final String scope;
+
+    @JsonProperty("sub")
+    private final String subject;
+
+    @JsonProperty("iss")
+    private final String issuer;
+
+    @JsonProperty("iat")
+    private final long issueDate;
+
+    @JsonProperty("jti")
+    private final String jsonWebTokenId;
+
+    @JsonProperty("error")
+    private final String error;
+
+    @JsonProperty("error_description")
+    private final String errorDescription;
+
+    @JsonProperty("message")
+    private final String message;
+
```

Comment:
Yeah this class is a bit confusing. It actually captures the response from ManagementPortal, rather than being a token. So i renamed it to OAuth2AccessTokenDetails. I also added documentation to clarify the intent of this class. The JWT token is contained in the `accessToken` field. I purposely did not include a JWT parser in this library. A client of this library might not need it (essentially you only need to provide the token to a protected resource), but if they do need it, they are free to choose a JWT library, they might already be using one. So information like `authorities`, if needed by the client, can be parsed from the JWT.